internal/services/apimanagement/api_management_diagnostic_resource.go

```diff
@@ -147,11 +153,18 @@ func resourceApiManagementDiagnosticCreateUpdate(d *pluginsdk.ResourceData, meta
 
 	parameters := diagnostic.DiagnosticContract{
 		Properties: &diagnostic.DiagnosticContractProperties{
-			LoggerId:            d.Get("api_management_logger_id").(string),
-			OperationNameFormat: pointer.To(diagnostic.OperationNameFormat(d.Get("operation_name_format").(string))),
+			LoggerId: d.Get("api_management_logger_id").(string),
 		},
 	}
 
+	if d.Get("identifier") == "applicationinsights" {
+		if operationNameFormat, ok := d.GetOk("operation_name_format"); ok {
+			parameters.Properties.OperationNameFormat = pointer.To(diagnostic.OperationNameFormat(operationNameFormat.(string)))
+		} else {
+			parameters.Properties.OperationNameFormat = pointer.To(diagnostic.OperationNameFormatName)
+		}
+	}
```

Comment:
This should also be wrapped in the 4.0 flag since after 4.0 we don't want to be sending `diagnostic.OperationNameFormatName` is `operation_name_format` is omitted?

Reply:
Fixed.